### PR TITLE
Eliminate most of incorrect SymbolTable usages

### PR DESCRIPTION
**Pull request by SvyatoslavScherbina**
Also do some cleanup.

### CODE REVIEW

File: backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/CAdapterGenerator.kt [hunk 1]
```diff
@@ -1012,9 +1050,9 @@ internal class CAdapterGenerator(
     private var functionIndex = 0
     fun nextFunctionIndex() = functionIndex++
 
-    internal val kGetTypeFuncType =
+    internal val kGetTypeFuncType get() =
```
Comment (homuroll) [inline backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/CAdapterGenerator.kt]: `lazy`?

Comment (SvyatoslavScherbina) [reply backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/CAdapterGenerator.kt]: Somewhat conflicts the idea of having `codegen` for a limited amount of time only.

File: backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/Context.kt [hunk 2]
```diff
@@ -203,6 +203,8 @@ internal class Context(config: KonanConfig) : KonanBackendContext(config) {
 
     lateinit var objCExport: ObjCExport
 
+    lateinit var cAdapterGenerator: CAdapterGenerator
```
Comment (olonho) [inline backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/Context.kt]: Hmm, this guy doesn't seem to belong to context.

File: backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/llvm/objcexport/ObjCExportCodeGenerator.kt [hunk 3]
```diff
@@ -160,29 +162,26 @@ internal class ObjCExportCodeGenerator(
...
+                    val superClass = it.superClassNotAny ?: objCClassForAny
+
+                    dataGenerator.emitEmptyClass(it.binaryName, superClass.binaryName)
+                    // Note: it is generated only to be visible for linker.
+                    // Methods will be added at runtime.
+
+                    createTypeAdapter(it, superClass)
+                }
+
+                is ObjCProtocolForKotlinInterface -> createTypeAdapter(it, superClass = null)
```
Comment (olonho) [inline backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/llvm/objcexport/ObjCExportCodeGenerator.kt]: Shalln't we have catching `else` here?

Comment (SvyatoslavScherbina) [reply backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/llvm/objcexport/ObjCExportCodeGenerator.kt]: This `when` is used as an expression and thus forced to be exhaustive.

**Pull request merged**